### PR TITLE
Adding official examples to tutorials

### PR DESCRIPTION
**Pull request by iriark01**

### CODE REVIEW

File: docs/introduction/tutorials_intro.md [hunk 1]
```diff
@@ -110,145 +114,158 @@ And the following tutorial:
...
+| API | Snippets | Tutorials | Official examples |
+| --- | --- | --- | --- |
+| Memory tracing (mbed_mem_trace) | [Memory tracing snippet](../apis/memory-apis.html#memory-tracing-example) | [Runtime memory tracing](../apis/runtime-memory-tracing.html) | |
+| MPU management (mpug_mgmt) | [MPU management snippet](../apis/mpu-management.html#example) | | |
+| MemoryPool | [MemoryPool snippet](../apis/memorypool.html#memorypool-example) | | |
+| Mbed statistics  (mbed_stats)| - [Memory statistics snippet](../apis/mbed-statistics.html#memory-statistics-example) <br> - [Thread statistics snippet](../apis/mbed-statistics.html#thread-statistics-example) <br> - [System information snippet](../apis/mbed-statistics.html#system-information-example) <br> - [CPU statistics snippet](../apis/mbed-statistics.html#cpu-statistics-example)|  | - [CPU statistics official example](https://github.com/ARMmbed/mbed-os-example-cpu-stats) </br> - [CPU usage official example](https://github.com/ARMmbed/mbed-os-example-cpu-usage) </br> - [Thread statistics official example](https://github.com/ARMmbed/mbed-os-example-thread-statistics)|
```
Comment (evedon) [inline docs/introduction/tutorials_intro.md]: CPU usage example is deprecated. You should replace with mbed-os-example-sys-info

File: docs/introduction/tutorials_intro.md [hunk 2]
```diff
@@ -4,104 +4,108 @@ The Arm Mbed ecosystem is expansive and offers many opportunities. In contrast t
...
+| EventQueue | - [Chaining events from more than one queue](../apis/eventqueue.html#eventqueue-example-chaining-events-from-more-than-one-queue)<br> - [Deferring from interrupt context](../apis/eventqueue.html#eventqueue-example-deferring-from-interrupt-contextl) <br> - [Posting events to the queue](../apis/eventqueue.html#eventqueue-example-posting-events-to-the-queue) <br> - [Shared event: deferring from interrupt context](../apis/eventqueue.html#shared-event-example-running-the-shared-queue-from-main)<br> - [Shared event: running the shared queue from main](../apis/eventqueue.html#shared-event-example-running-the-shared-queue-from-main)<br> - [Static event queue snippet](../apis/eventqueue.html#static-eventqueue-example-posting-user-allocated-events-to-the-static-queue) | [EventQueue tutorial](../apis/scheduling-tutorials.html)|
+| UserAllocatedEvent | [Static event queue snippet](../apis/userallocatedevent.html#static-eventqueue-example-posting-user-allocated-events-to-the-queue) | |
 
 ## Serial (UART) Driver APIs
 
-| API | Examples  | Hello, World |
+| API | Snippets  | Hello, World |
```
Comment (evedon) [inline docs/introduction/tutorials_intro.md]: There are no Hello World example so 2nd column can be removed

File: docs/introduction/tutorials_intro.md (hunk 1 above)
Comment (evedon) [inline docs/introduction/tutorials_intro.md]: The column title already says that the example if official so it does not need to be repeated

File: docs/introduction/tutorials_intro.md [hunk 3]
```diff
@@ -110,145 +114,158 @@ And the following tutorial:
...
-| Span | [Span example](../apis/span.html#span-example) |
+| API | Snippets | Official examples |
+| --- | --- | --- |
+| Assert | [Assert snippet](../apis/other-platform-apis.html#assert-example) | |
+| ATCmdParser | [ATCmdParser snippet](../apis/atcmdparser.html#atcmdparser-examples) | |
+| Callback | [Serial passthrough snippet](../apis/callback.html#serial-passthrough-example-with-callbacks)<br> [Thread snippet](../apis/callback.html#thread-example-with-callbacks)<br> [Sonar snippet](../apis/callback.html#sonar-example) | |
+| CircularBuffer | [CircularBuffer snippet](../apis/circularbuffer.html#circularbuffer-example) | |
+| CriticalSectionLock | [CriticalSectionLock snippet](../apis/criticalsectionlock.html#criticalsectionlock-example) | |
+| Debug | [Debug snippet](../apis/debug.html#debug-example) | |
+| Error handling | [Error handling snippet](../apis/error-handling.html#error-handling-example)<br> [Crash reporting snippet](../apis/error-handling.html#crash-reporting-example) | [Error handling official example](https://github.com/ARMmbed/mbed-os-example-error-handling) |
```
Comment (evedon) [inline docs/introduction/tutorials_intro.md]: We also have https://github.com/ARMmbed/mbed-os-example-crash-reporting

File: docs/introduction/tutorials_intro.md [hunk 4]
```diff
@@ -110,145 +114,158 @@ And the following tutorial:
...
-| NFCController | [NFCController example](../apis/nfccontroller.html#nfccontroller-example) |
-| NFCEEPROM | [NFCEEPROM example](../apis/nfc-eeprom.html#nfceeprom-example) |
-| SimpleMessageParser | [SimpleMessageParser example](../apis/simplemessageparser.html#simplemessageparser-example) |
+| MessageBuilder | [MessageBuilder snippet](../apis/nfc-apis.html#messagebuilder-example) |
+| MessageParser | [MessageParser snippet](../apis/messageparser.html#messageparser-example) || Mesh | | [Mesh tutorial](../apis/connectivity-tutorials.html) <br>  [Light control tutorial](../apis/light-control.html) |
+| NFCController | [NFCController snippet](../apis/nfccontroller.html#nfccontroller-example) |
+| NFCEEPROM | [NFCEEPROM snippet](../apis/nfc-eeprom.html#nfceeprom-example) |
+| SimpleMessageParser | [SimpleMessageParser snippet](../apis/simplemessageparser.html#simplemessageparser-example) |
 
+And the [official examples repository](https://github.com/ARMmbed/mbed-os-example-nfc).
```
Comment (evedon) [inline docs/introduction/tutorials_intro.md]: The two NFC examples should be linked in the tables above. We have
- https://github.com/ARMmbed/mbed-os-example-nfc/blob/master/NFC_SmartPoster/README.md
- https://github.com/ARMmbed/mbed-os-example-nfc/tree/master/NFC_EEPROM/README.md

File: docs/introduction/tutorials_intro.md [hunk 5]
```diff
@@ -110,145 +114,158 @@ And the following tutorial:
...
-| API | Examples | Tutorials |
+| API | Snippets | Tutorials |
 | --- | --- | --- |
-| DNS Resolver | [DNS Resolver example](../apis/dns-apis.html#dns-resolver-example) | |
+| DNS Resolver | [DNS Resolver snippet](../apis/dns-apis.html#dns-resolver-example) | |
 
 ## Bluetooth (BLE) APIs
 
-We have placed all of our BLE examples in a single [GitHub repository](https://github.com/ARMmbed/mbed-os-example-ble).
+We have placed all of our official BLE examples in a single [GitHub repository](https://github.com/ARMmbed/mbed-os-example-ble).
```
Comment (evedon) [inline docs/introduction/tutorials_intro.md]: The BLE examples should be inidividually linked.

**Review by evedon — COMMENTED** [review]
I started to put individual comments but I realised that I really don't like the page.
We are losing the essential, useful information i.e. the official examples and the tutorials by cluttering with code snippets and hello world examples. The introduction announces what this page should be about but then it get lost in the details. The code snippets are _already_ in the APIs section so I don't see why we should add them here.

The second point is that it would be very useful to list all the BLE, NFC, TLS examples instead of giving a link to the main repo.

**Comment by iriark01** [comment]
Okay, I have no idea why there are so many empty lines in each cell. While I figure that out, please take a look: https://os.mbed.com/docs/mbed-os/tutorials-list-update/tutorials/index.html

**Comment by iriark01** [comment]
Sorted out the weird table lines; ready for review:
https://os.mbed.com/docs/mbed-os/tutorials-list-update/tutorials/index.html
FYI @Wenchou01

File: docs/introduction/tutorials_intro.md [hunk 6]
```diff
@@ -1,254 +1,29 @@
...
+| Drivers | [Alarm](../apis/drivers-tutorials.html) | |
+| Network interfaces | - [Mesh tutorial](./apis/connectivity-tutorials.html) <br/> - [LoRaWAN usage](../apis/lorawan-usage.html) <br/> - [Building a private loRa network](../apis/LoRa-tutorial.html) <br/> - [Nanostack border router](https://github.com/ARMmbed/nanostack-border-router) | - [Cellular](https://github.com/ARMmbed/mbed-os-example-cellular) <br/> - [LoRaWAN](https://github.com/ARMmbed/mbed-os-example-lorawan) <br/> - [Mesh minimal](https://github.com/ARMmbed/mbed-os-example-mesh-minimal) (also available in the [light control tutorial](../apis/light-control.html) <br/> - [Wi-Fi ](https://github.com/ARMmbed/mbed-os-example-wifi) |
```
Comment (evedon) [inline docs/introduction/tutorials_intro.md]: A close parenthesis is missing
```suggestion
| Network interfaces | - [Mesh tutorial](./apis/connectivity-tutorials.html) <br/> - [LoRaWAN usage](../apis/lorawan-usage.html) <br/> - [Building a private loRa network](../apis/LoRa-tutorial.html) <br/> - [Nanostack border router](https://github.com/ARMmbed/nanostack-border-router) | - [Cellular](https://github.com/ARMmbed/mbed-os-example-cellular) <br/> - [LoRaWAN](https://github.com/ARMmbed/mbed-os-example-lorawan) <br/> - [Mesh minimal](https://github.com/ARMmbed/mbed-os-example-mesh-minimal) (also available in the [light control tutorial](../apis/light-control.html)) <br/> - [Wi-Fi ](https://github.com/ARMmbed/mbed-os-example-wifi) |
```

File: docs/introduction/tutorials_intro.md [hunk 7]
```diff
@@ -1,254 +1,29 @@
...
+| Network socket | | - [Sockets](https://github.com/ARMmbed/mbed-os-example-sockets) <br/> - [Socket statistics](https://github.com/ARMmbed/mbed-os-example-socket-stats) |
+| NFC | | - [Smart Poster](https://github.com/ARMmbed/mbed-os-example-nfc/blob/master/NFC_SmartPoster/README.md) <br/> - [EEPROM](https://github.com/ARMmbed/mbed-os-example-nfc/blob/master/NFC_EEPROM/README.md)|
+| Network secure socket | | [TLS socket](https://github.com/ARMmbed/mbed-os-example-tls-socket) |
+| Platform | - [Application flow control](../apis/platform-tutorials.html) <br/> - [Power optimization](../apis/power-optimization.html) | - [Thread statistics](https://github.com/ARMmbed/mbed-os-example-thread-statistics) <br/> - [CPU statistics](https://github.com/ARMmbed/mbed-os-example-cpu-stats) <br/> - [Error handling](https://github.com/ARMmbed/mbed-os-example-error-handling) |
```
Comment (evedon) [inline docs/introduction/tutorials_intro.md]: Other platform tutorials: 
- https://github.com/ARMmbed/mbed-os-example-sys-info
- https://github.com/ARMmbed/mbed-os-example-crash-reporting

File: docs/introduction/tutorials_intro.md [hunk 8]
```diff
@@ -1,254 +1,29 @@
-# Arm Mbed tutorials
+# Tutorials and official examples
 
-The Arm Mbed ecosystem is expansive and offers many opportunities. In contrast to other sections of the documentation, which provide background reference material, this section provides instructions for specific tasks you may wish to perform. This section contains two formats of documentation: tutorials and examples. Tutorials include step-by-step guidance, and examples are code snippets you can use as a starting point for your application or as a reference on how to use a particular API.
+This page is an index of all tutorials and official examples.
```
Comment (evedon) [inline docs/introduction/tutorials_intro.md:3]: This page does not list blinky and blinky bare metal but that's probably okay as they are linked in other pages.

File: docs/introduction/tutorials_intro.md [hunk 9]
```diff
@@ -258,38 +33,31 @@ These tutorials show you how to install, export a project to and start a debuggi
...
+## Testing
+
+- [Test your applications or Mbed OS with Greentea](../debug-test/greentea-for-testing-applications.html).
+- [Using the Icetea test framework](../debug-test/icetea.html) and [test applications](../debug-test/icetea-test-applications.html).
 
 ## Bootloader
 
-A [tutorial for creating and using a bootloader](../program-setup/creating-and-using-a-bootloader.html), and [an example bootloader](https://github.com/ARMmbed/mbed-os-example-bootloader).
+- [A tutorial for creating and using a bootloader](../program-setup/creating-and-using-a-bootloader.html).
+- [An official example](https://github.com/ARMmbed/mbed-os-example-bootloader).
```
Comment (evedon) [inline docs/introduction/tutorials_intro.md]: mbed-os-example-bootloader is an official example of how a bootloader could be implemented.
The Pelion bootloader is https://github.com/ARMmbed/mbed-bootloader

File: docs/introduction/tutorials_intro.md [hunk 10]
```diff
@@ -258,38 +33,31 @@ These tutorials show you how to install, export a project to and start a debuggi
...
+
+## Connecting to the cloud
 
-## Optimizing
+[Pelion Device Management](https://github.com/ARMmbed/mbed-os-example-pelion)
```
Comment (evedon) [inline docs/introduction/tutorials_intro.md:58]: mbed-os-example-pelion is a lightweigth simple example.
The Pelion example is https://github.com/armmbed/mbed-cloud-client-example

**Review by evedon — COMMENTED** [review]
This is great @iriark01 
This page is so much better now

File: docs/introduction/tutorials_intro.md [hunk 11]
```diff
@@ -258,38 +33,31 @@ These tutorials show you how to install, export a project to and start a debuggi
 
 - [BBC micro:bit, pyOCD and GDB debugging tutorial](../debug-test/debug-microbit.html).
 - [Troubleshooting common issues](../debug-test/troubleshooting-common-issues.html).
-- [Analyzing Mbed OS crash dump tutorial](../debug-test/analyzing-mbed-os-crash-dump.html).
+- [Analyzing Mbed OS crash dump tutorial](../debug-test/analyzing-mbed-os-crash-dump.html) and [crash reporting official example](https://github.com/ARMmbed/mbed-os-example-crash-reporting)
```
Comment (iriark01) [inline docs/introduction/tutorials_intro.md:36]: @evedon Crash Reporting is here. Where should the Sys Info one go?

File: docs/introduction/tutorials_intro.md (hunk 8 above)
Comment (iriark01) [reply docs/introduction/tutorials_intro.md:3]: I did at first link to the quick starts but it just ends up being a third instance of linking to them, and they're already a top-level item, so - as you said - I decided it's a bit much.

File: docs/introduction/tutorials_intro.md (hunk 9 above)
Comment (iriark01) [reply docs/introduction/tutorials_intro.md]: I think I'll leave the Pelion docs to link to their bootloader.

File: docs/introduction/tutorials_intro.md (hunk 10 above)
Comment (iriark01) [reply docs/introduction/tutorials_intro.md:58]: Never sure which version of the OS they use in their example; I'll keep using just the Mbed OS official

Comment (iriark01) [reply docs/introduction/tutorials_intro.md:58]: BTW, do you know whether the AWS example can be linked? The link I have - https://github.com/ARMmbed/mbed-os-example-aws - 404s

File: docs/introduction/tutorials_intro.md (hunk 11 above)
Comment (evedon) [reply docs/introduction/tutorials_intro.md:36]: Under platform

File: docs/introduction/tutorials_intro.md (hunk 10 above)
Comment (evedon) [reply docs/introduction/tutorials_intro.md:58]: Not yet

**Comment by iriark01** [comment]
Merging, because I added the examples Evelyne needed and that was the only merge blocker for her

**Pull request merged**